Modix.Bot/Behaviors/PromotionPollBehavior/PromotionDialogBehavior.cs

```diff
@@ -0,0 +1,211 @@
+
+        public async Task HandleNotificationAsync(PromotionActionCreatedNotification notification,
+            CancellationToken cancellationToken = default)
+        {
+            if (cancellationToken.IsCancellationRequested)
+                return;
+
+            switch (notification.Data.Type)
+            {
+                case PromotionActionType.CampaignCreated:
+                    await CreatePromoDialog(notification.Data);
+                    break;
+                case PromotionActionType.CampaignClosed:
+                    await DeletePromoDialog(notification.Data);
+                    break;
+            }
+        }
+
+        public async Task HandleNotificationAsync(ReactionAddedNotification notification,
+            CancellationToken cancellationToken = default)
+        {
+            var reaction = notification.Reaction.Emote;
+            var user = (IGuildUser)notification.Reaction.User.Value;
+            var message = await notification.Message.GetOrDownloadAsync();
+
+            if (cancellationToken.IsCancellationRequested
+                || notification.Reaction.User.Value.IsBot
+                || message.Embeds.Count < 1)
+                return;
+
+            if (_memoryCache.TryGetValue(message.Id, out CachedPromoDialog poll)
+            && ValidateVoteReaction(reaction))
+            {
+                string voteError = null;
+                var sentiment = reaction.Name == _approveEmoji.Name ? PromotionSentiment.Approve : PromotionSentiment.Oppose;
+
+                await message.RemoveReactionAsync(notification.Reaction.Emote, notification.Reaction.User.Value);
+
+                try
+                {
+                    await _authorizationService.OnAuthenticatedAsync(user);
+                    await _promotionsService.UpdateOrAddComment(poll.CampaignId, sentiment, "This vote was cast from Discord");
+                }
+                catch (Exception e)
+                {
+                    voteError = e.Message;
+                }
+
+                var campaign = await _promotionCampaignRepository.GetCampignSummaryByIdAsync(poll.CampaignId);
+
+                await message.ModifyAsync(m =>
+                    m.Embed = BuildPollEmbed(campaign, campaign.ApproveCount, campaign.OpposeCount, voteError).Build());
+
+                await Task.Delay(ErrorRemoveDelay, cancellationToken);
+
+                if(voteError != null)
+                    await message.ModifyAsync(m =>
+                        m.Embed = BuildPollEmbed(campaign, campaign.ApproveCount, campaign.OpposeCount).Build());
+            }
+        }
+
+        private async Task CreatePromoDialog(PromotionActionCreationData campaign)
+        {
+            if (!await _designatedChannelService.AnyDesignatedChannelAsync(campaign.GuildId,
+                DesignatedChannelType.PromotionInterface))
+                return;
+
+            var pollChannel = await GetPollChannel(_discordSocketClient.GetGuild(campaign.GuildId));
```

Comment:
I feel like we talked about this briefly, and maybe we had a reason for this, but I forget...

Don't assume there's only one designated channel.  `IDesignatedChannelService.GetDesignatedChannelsAsync()` returns a collection for a reason, If the users have designated multiple channels, it's your job to use all of them.

Also, use `IDesignatedChannelService.SendToDesignatedChannelsAsync()`

Also, if you make this change, that means you're gonna have to change your caching strategy to keep a list of message IDs for each campaign ID.

Reply:
This is one of those things that i plan on adding with a few other improvements after there is some feedback from people in the server, this along with time to close and some basic participation  are whats next

Reply:
Fair enough.